connexion/json_schema.py

```diff
@@ -2,21 +2,70 @@
 Module containing all code related to json schema validation.
 """
 
+import contextlib
+import io
+import os
 import typing as t
+import urllib.parse
+import urllib.request
 from collections.abc import Mapping
 from copy import deepcopy
 
+import requests
+import yaml
 from jsonschema import Draft4Validator, RefResolver
 from jsonschema.exceptions import RefResolutionError, ValidationError  # noqa
 from jsonschema.validators import extend
-from openapi_spec_validator.handlers import UrlHandler
 
 from .utils import deep_get
 
+
+class ExtendedSafeLoader(yaml.SafeLoader):
+    """Extends the yaml SafeLoader to coerce all keys to string so the result is valid json."""
```

Comment:
Why do we need to do this? Shouldn't the loaded file be json as well if it adheres to OpenAPI? Perhaps I'm simply overlooking something.

Reply:
The loaded file can be a `yaml` file as well, which allows integer keys. According to the OpenAPI spec, keys should indeed be strings so the `yaml` is a representation of valid `json`, and we shouldn't need this. However `openapi-spec-validator` allowed this, and I tried to provide a drop-in replacement so kept the same behavior.

We could mark this for deprecation in the next major version though.